dist/performance.js

```diff
@@ -11,34 +23,106 @@ function getLcpElement() {
+            nodeName: element?.nodeName,
             url,
             size,
             loadTime,
             renderTime,
             attributes,
+            boundingClientRect: element?.getBoundingClientRect().toJSON(),
+            naturalWidth: element?.naturalWidth,
+            naturalHeight: element?.naturalHeight,
+            styles
+        };
+    });
+}
+
+function getAttributes(element) {
+    if (!element) {
+        return null;
+    }
+
+    return Array.from(element.attributes).map(attr => {
+        return {
+            name: attr.name,
+            value: attr.value
         };
     });
 }
 
+function getComputedStyles(element, properties) {
+    if (!element) {
+        return null;
+    }
+
+    const styles = getComputedStyle(element);
+    return Object.fromEntries(properties.map(prop => ([prop, styles.getPropertyValue(prop)])));
+}
+
+function summarizeLcpElement(element) {
+    if (!element) {
+        return null;
+    }
+
+    const nodeName = element.nodeName;
+    const attributes = getAttributes(element);
+
+    return {
+        nodeName,
+        attributes
+    }
+}
+
 function getWebVitalsJS() {
     const webVitalsJSPattern = /(8999999999999[\s\S]+1e12[\s\S]+(largest-contentful-paint|first-input|layout-shift)|(largest-contentful-paint|first-input|layout-shift)[\s\S]+8999999999999[\s\S]+1e12)/m;
     return response_bodies.filter(har => {
         return webVitalsJSPattern.test(har.response_body);
     }).map(har => har.url);
 }
 
-return Promise.all([getLcpElement()]).then(lcp_elem_stats => {
+return Promise.all([getLcpElement()]).then(([lcp_elem_stats]) => {
+    const lcpUrl = lcp_elem_stats.url;
+    const rawDoc = getRawHtmlDocument();
+    let isLcpDiscoverable = null;
+    let isLcpPreloaded = null;
+    let responseObject = null;
+    let rawLcpElement = null;
+    const isLcpExternalResource = lcpUrl != '';
+    if (isLcpExternalResource) {
+        // Check if LCP resource reference is in the raw HTML (as opposed to being injected later by JS)
+        rawLcpElement = Array.from(rawDoc.querySelectorAll('picture source, img')).find(i => {
+            let src = i.src;
+            if (i.nodeName == 'SOURCE') {
+                src = new URL(i.srcset, location.href).href;
+            }
+
+            return src == lcpUrl;
+        });
+        isLcpDiscoverable = !!rawLcpElement;
+        isLcpPreloaded = !!Array.from(rawDoc.querySelectorAll('link')).find(link => {
+            return link.rel == 'preload' && link.href == lcpUrl;
+        });
+        responseObject = response_bodies.find(r => {
+            return r.url == lcpUrl;
+        });
+        if (responseObject) {
+            // Don't write the response body to custom metrics.
+            responseObject.response_body = undefined;
+        }
+    }
+
     return {
         lcp_elem_stats,
+        raw_lcp_element: summarizeLcpElement(rawLcpElement),
+        lcp_resource: responseObject,
+        is_lcp_discoverable: isLcpDiscoverable,
```

Comment:
Thinking about naming conventions - if I came across this in a database a few years from now I might mistake this for 'Is there an LCP element at all?' instead of 'Is there an LCP element in the raw HTML?'. That said, I can't really think of a better name...does `is_lcp_dynamically_loaded` have less room for confusion?

Reply:
I like that. How about `is_lcp_statically_discoverable` or `is_lcp_speculatively_discoverable`?

Reply:
Sounds good - if I had to pick I would vote for `statically`!